src/foremast/utils/dns.py

```diff
@@ -91,6 +91,43 @@ def update_dns_zone_record(env, zone_id, **kwargs):
 
     LOG.debug('Route53 JSON Response: \n%s', pformat(response))
 
+def delete_existing_cname(env, zone_id, **kwargs):
+    """Function to delete an existing CNAME record. This is
+    used when updating to multi-region for deleting old records.
+    The record can not just be upserted since it changes types.
+
+    Args:
+        env (str): Deployment environment.
+        zone_id (str): Route53 zone id.
+
+    Keyword Args:
+        dns_name (str): FQDN of application's dns entry to add/update.
+
+    """
+    oldrecord = None
+    recordname = kwargs.get('dns_name')
+    client = boto3.Session(profile_name=env).client('route53')
+    response = client.list_resource_record_sets(
+        HostedZoneId=zone_id
+    )
+    for item in response['ResourceRecordSets']:
+        if item['Name'][:-1] == recordname and item['Type'] == 'CNAME':
```

Comment:
[:-1] is be cause there is a period at the end of `item["Name"]`. I could just use start with I guess but this felt more accurate to me